### PR TITLE
Remove some Win9x legacy

### PR DESCRIPTION
**Pull request by nir9**
Problem: The source tree contains some Win9x legacy.

Solution: Remove some of the legacy

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -285,10 +285,10 @@ dist/$(COMMENT_RT): dist/comment
 	echo "Vim - Vi IMproved - v$(VDOT) runtime files for MS-DOS and MS-Windows" > dist/$(COMMENT_RT)
 
 dist/$(COMMENT_W32): dist/comment
-	echo "Vim - Vi IMproved - v$(VDOT) binaries for MS-Windows NT/95" > dist/$(COMMENT_W32)
+	echo "Vim - Vi IMproved - v$(VDOT) binaries for MS-Windows NT" > dist/$(COMMENT_W32)
```
Comment (k-takata) [inline Makefile]:
```suggestion
	echo "Vim - Vi IMproved - v$(VDOT) binaries for MS-Windows" > dist/$(COMMENT_W32)
```
(Same for other part.)

Comment (nir9) [reply Makefile]: fixed

**Comment by chrisbra** [comment]
I am slightly worried that removing the edge-case handling of `mch_rename` regresses unneccessarily. Can you please keep this, if this is still an issue that occurs on recent Windows versions?

**Comment by nir9** [comment]
@chrisbra Fixed and adjusted the comment to only mention the edge case that is relevant for modern Windows versions (and is handled by the function logic)

**Comment by chrisbra** [comment]
thanks!

**Pull request closed without merging**